### ISSUE TITLE
Asset/Resource system

### ISSUE THREAD

**Issue by peter-r-g** [xezno/mocha#33]
Right now the asset/resource system is a bit all over the place. Firstly it currently has two names. In Mocha.Common it is referred to as Resources; Whereas in Mocha.Engine and the AssetCompiler it is referred to as Assets. As well, asset logic is spread across multiple projects. Ideally, we should have all of the assets in a single place that contains all of their functionality.

I suggest that we collect all of the asset-related code and put it under a single `Mocha.Assets` project. That way we have all of the required functionality in one place. With that, we also won't have to reference a tool for compiling logic.

**Comment by xezno** [comment]
Agreed. I think everything under `ResourceType` should also be renamed to `AssetType` too for consistency's sake, I much prefer the shorter `Asset` naming.
